### PR TITLE
Update ExaGO to use slim petsc build

### PR DESCRIPTION
**Pull request by ashermancinelli**
Update ExaGO to use slim petsc build, add myself and CameronRutherford as maintainers.

### CODE REVIEW

**Review by cameronrutherford — APPROVED** [review]
LGTM

File: var/spack/repos/builtin/packages/exago/package.py [hunk 1]
```diff
@@ -56,7 +57,7 @@ class Exago(CMakePackage, CudaPackage):
     depends_on('hiop+mpi', when='+hiop+mpi')
 
     # Require PETSc < 3.15 per ExaGO issue #199
-    depends_on('petsc@3.13:3.14', when='+petsc')
+    depends_on('petsc@3.13:3.14~hypre~hdf5~metis~superlu-dist', when='+petsc')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/exago/package.py:60]: Does this mean that I can't use at all `petsc+hypre` to build this package?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a doubt on whether `petsc~hypre~hdf5~metis~superlu-dist` is a preference or a requirement for this software. If it is a preference it shouldn't be expressed in a `depends_on` but in `packages.yaml`.

**Comment by ashermancinelli** [comment]
@alalazo You're right, this is a preference. We can just use the packages.yaml for this.

**Comment by alalazo** [comment]
Thanks for the quick reply

**Pull request closed without merging**